src/backend/distributed/commands/multi_copy.c

```diff
@@ -1716,8 +1726,8 @@ CopyFlushOutput(CopyOutState cstate, char *start, char *pointer)
  * a distributed table.
  */
```

Comment:
Might be worth adding some comments on the difference between `INSERT..SELECT` and `COPY`, and we could maybe add the comment here while explaining why we pass `partitionColumnIndex `?